### PR TITLE
Add better support for the new RC API

### PR DESCRIPTION
**Pull request by christophe-papazian**
## Motivation

We need to be able to 

- delete RC files (for example, one click deactivation is working like that)
- keep track of RC states between tests of a same scenario

Remote config works like a permanent state represented as a set of files (configurations) that we can modify or delete. This PR changes the API to reflect that. 

Please remember that each scenario using remote config tests share a single state among all tests of the scenario. No assumption should be made on the tests order, so each setup should set the desired state with `reset` and `set_config` at first (at test or class level).

## Changes

- refactor the api by exposing a single `rc_state` object
- rename `add_client_config` to `set_config`
- add `del_config` and `reset` to delete RC files in the current state
- automatically keep track of config and files version to ensure the next send is taken into account as expected
- allow empty list of files to be sent to reset everything and wait for confirmation (using global version watching)
- update documentation
- update all previous tests using the new API
- Add a more complex test enabling and disabling ASM 4 times in a row, waiting and testing at each step.
- Update all manifests

APPSEC-54105

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: utils/_remote_config.py [hunk 1]
```diff
@@ -32,23 +32,29 @@ def _post(path: str, payload) -> None:
...
+    and the values contain the actual state for each file:
+
+    1. a config state acknowledging the config
+    2. else if not acknowledged, the last config state received
+    3. if no config state received, then a hardcoded one with apply_state=UNKNOWN
+
+    Arguments:
+        wait_for_acknowledge_status
+            If True, waits for the config to be acknowledged by the library.
+            Else, only wait for the next request sent to /v0.7/config
```
Comment (cbeauchesne) [inline utils/_remote_config.py:57]: Could you ask a description of `command_version`, and how it works ?

File: utils/_remote_config.py [hunk 2]
```diff
@@ -252,25 +265,44 @@ class RemoteConfigCommand:
...
-
-    def add_client_config(self, path, config, config_file_version=None) -> ClientConfig:
-        client_config = ClientConfig(path=path, config=config, config_file_version=config_file_version)
-        self.targets.append(client_config)
-
-        return client_config
+    _store: dict[str, "RemoteConfigCommand"] = {}
+
+    def __new__(cls, expires=None) -> "RemoteConfigCommand":
+        scenario = context.scenario.name
```
Comment (cbeauchesne) [inline utils/_remote_config.py:275]: You cannot have two different scenario during the same process.

**Review by cbeauchesne — COMMENTED** [review]
Reviewed by zoom, some changes to come :)

File: utils/_remote_config.py [hunk 3]
```diff
@@ -292,17 +327,23 @@ def to_payload(self, deserialized=False):
...
 
-    def send(self, *, wait_for_acknowledged_status: bool = True) -> dict[str, dict[str, Any]]:
+    def apply(self, *, wait_for_acknowledged_status: bool = True) -> dict[str, dict[str, Any]]:
+        self.version += 1
         command = self.to_payload()
-        return send_command(command, wait_for_acknowledged_status=wait_for_acknowledged_status)
+        return send_state(
+            command, wait_for_acknowledged_status=wait_for_acknowledged_status, state_version=self.version
+        )
+
```
[the remark this answers is not in the document]
Comment (christophe-papazian) [reply utils/_remote_config.py:347]: There is already one ! I'm not sure what you mean there.

[the remark this answers is not in the document]
Comment (cbeauchesne) [reply utils/_remote_config.py:347]: haha, too fast review

**Review by cbeauchesne — APPROVED** [review]

**Comment by cbeauchesne** [comment]
❤️

**Pull request merged**